crates/sparrow-runtime/src/metadata/raw_metadata.rs

```diff
@@ -21,43 +41,117 @@ pub struct RawMetadata {
-    pub fn try_from_raw_schema(raw_schema: SchemaRef) -> anyhow::Result<Self> {
+    pub fn from_raw_schema(raw_schema: SchemaRef) -> Self {
         // Convert the raw schema to a table schema.
         let table_schema = convert_schema(raw_schema.as_ref());
 
-        Ok(Self {
+        Self {
             raw_schema,
             table_schema,
-        })
+        }
+    }
+
+    /// Create a `RawMetadata` from a parquet string path and object store registry
+    async fn try_from_parquet(
+        path: &str,
+        object_store_registry: Option<&ObjectStoreRegistry>,
+    ) -> error_stack::Result<Self, Error> {
+        let object_store_url = ObjectStoreUrl::from_str(path)
+            .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
+        let object_store_key = object_store_url
+            .key()
+            .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
+        match (object_store_key, object_store_registry) {
+            (ObjectStoreKey::Local, _) => {
+                let path = object_store_url
+                    .path()
+                    .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?
+                    .to_string();
+                let path = format!("/{}", path);
+                let path = std::path::Path::new(&path);
+                Self::try_from_parquet_path(path)
+                    .into_report()
+                    .change_context_lazy(|| Error::ReadSchemaIssue)
+            }
+            (_, None) => error_stack::bail!(Error::NoObjectStoreRegistry),
+            (_, Some(object_store_registry)) => {
+                let download_file = NamedTempFile::new().map_err(|_| Error::DownloadIssue)?;
+                object_store_url
+                    .download(object_store_registry, download_file.path().to_path_buf())
+                    .await
+                    .change_context_lazy(|| Error::DownloadIssue)?;
+                Self::try_from_parquet_path(download_file.path())
+                    .into_report()
+                    .change_context_lazy(|| Error::ReadSchemaIssue)
+            }
+        }
+    }
+
+    /// Create a `RawMetadata` from a CSV string path and object store registry
+    async fn try_from_csv(
+        path: &str,
+        object_store_registry: Option<&ObjectStoreRegistry>,
+    ) -> error_stack::Result<Self, Error> {
+        let object_store_url = ObjectStoreUrl::from_str(path)
+            .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
+        let object_store_key = object_store_url
+            .key()
+            .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
+        match (object_store_key, object_store_registry) {
+            (ObjectStoreKey::Local, _) => {
+                let path = object_store_url
+                    .path()
+                    .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?
+                    .to_string();
+                let path = format!("/{}", path);
```

Comment:
This probably deserves some documentation. What happens if the `path` already has a prefix `/`?

Reply:
Updated with some docs.